test/irqbalance.bats

```diff
@@ -0,0 +1,198 @@
+		skip "requires default IRQ smp affinity (not banned CPUs)"
+	fi
+	[ -f "$CONFIGLET" ] && rm -f "$CONFIGLET"
+
+	[ -f "$BANNEDCPUS_CONF" ] && rm -f "$BANNEDCPUS_CONF"
+
+	local expected_banned_cpus
+	expected_banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+	if [ -z "$expected_banned_cpus" ]; then
+		echo "error: unable to read 'IRQBALANCE_BANNED_CPUS' from ${IRQBALANCE_CONF}"
+		return 1
+	fi
+
+	# when
+	OVERRIDE_OPTIONS="--irqbalance-config-file ${IRQBALANCE_CONF}" start_crio
+
+	# then
+	if [ ! -f "$BANNEDCPUS_CONF" ]; then
+		echo "error: ${BANNEDCPUS_CONF} file should have been created by CRI-o"
+		return 2
+	fi
+
+	local banned_cpus
+	banned_cpus=$(cat "$BANNEDCPUS_CONF")
+
+	[ "$expected_banned_cpus" == "$banned_cpus" ]
+}
+
+@test "irqbalance cpu ban list restore - default" {
+	# given
+	if ! grep -Eq '^[1,3,7,f]{1,}$' /proc/irq/default_smp_affinity; then
+		skip "requires default IRQ smp affinity (not banned CPUs)"
+	fi
+	[ -f "$CONFIGLET" ] && rm -f "$CONFIGLET"
+
+	echo "IRQBALANCE_BANNED_CPUS=\"0\"" > "$IRQBALANCE_CONF"
+
+	local banned_cpus_for_conf
+	banned_cpus_for_conf=$(cat /proc/irq/default_smp_affinity)
+	echo "$banned_cpus_for_conf" > "$BANNEDCPUS_CONF"
+
+	# when
+	OVERRIDE_OPTIONS="--irqbalance-config-file ${IRQBALANCE_CONF}" start_crio
+
+	# then
+	local banned_cpus
+	banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+
+	[ "$banned_cpus_for_conf" == "$banned_cpus" ]
+}
+
+# disable restore file, check it does NOT clear the irqbalance config
+@test "irqbalance cpu ban list restore - disable" {
+	# given
+	if ! grep -Eq '^[1,3,7,f]{1,}$' /proc/irq/default_smp_affinity; then
+		skip "requires default IRQ smp affinity (not banned CPUs)"
+	fi
+	[ -f "$CONFIGLET" ] && rm -f "$CONFIGLET"
+
+	local expected_banned_cpus
+	expected_banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+	if [ -z "$expected_banned_cpus" ]; then
+		echo "error: unable to read 'IRQBALANCE_BANNED_CPUS' from ${IRQBALANCE_CONF}"
+		return 1
+	fi
+
+	[ -f "$BANNEDCPUS_CONF" ] && rm -f "$BANNEDCPUS_CONF"
+
+	# when
+	OVERRIDE_OPTIONS="--irqbalance-config-file ${IRQBALANCE_CONF} --irqbalance-config-restore-file disable" start_crio
+
+	# then
+	local banned_cpus
+	banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+	if [ -z "$banned_cpus" ]; then
+		echo "error: unable to read 'IRQBALANCE_BANNED_CPUS' from ${IRQBALANCE_CONF}"
+		return 2
+	fi
+
+	[ "$expected_banned_cpus" == "$banned_cpus" ]
+}
+
+# explicit restore file, check it does SET the irqbalance config accordingly
+@test "irqbalance cpu ban list restore - explicit file" {
+	# given
+	if ! grep -Eq '^[1,3,7,f]{1,}$' /proc/irq/default_smp_affinity; then
+		skip "requires default IRQ smp affinity (not banned CPUs)"
+	fi
+	[ -f "$CONFIGLET" ] && rm -f "$CONFIGLET"
+
+	[ -f "$BANNEDCPUS_CONF" ] && rm -f "$BANNEDCPUS_CONF"
```

Comment:
I'm thinking if is better to keep this file and create a new temporary one (like you already doing) to emphasize the fact we are consuming data from the file given explicitely, not from the hardcoded old default path.

Reply:
Maybe we can  check that this file is not even created to emphasize the fact that this file is not handled in any possible way by the code.

Reply:
I have added the check to ensure the default file is not even created after the cri-o run, so it is completely ignored.